include/swift/Basic/PrintDiagnosticNamesMode.h

```diff
@@ -0,0 +1,22 @@
+//===--- PrintDiagnosticNamesMode.h -----------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_PRINTDIAGNOSTICNAMESMODE_H
+#define SWIFT_BASIC_PRINTDIAGNOSTICNAMESMODE_H
+
+namespace swift {
+
+enum class PrintDiagnosticNamesMode { None, Identifier, Group };
```

Comment:
I'd appreciate some comments on the enum + cases, since this enum is pretty far away from the command-line flags that set its values.